bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -58,51 +61,57 @@ def optimizedparam(sampleval, nfold=2):
 if args.make_signal_kde and args.make_template_kde:
     raise Exception("Choose only one option out of --make-signal-kde and --make-template-kde")
 
-#calling template data and constructing template_kde_hdf file over template params
-template_file = h5py.File(args.template_file, 'r')
-mass1_n = template_file['mass1'][:]
-mass2_n = template_file['mass2'][:]
-spin1z_n = template_file['spin1z'][:]
-spin2z_n = template_file['spin2z'][:]
-chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
-mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
-eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+#calling template data and constructing template_kde_hdf_file over same template params
+temp_file = h5py.File(args.template_file, 'r')
+approximant = temp_file['approximant'][:]
+f_lower = temp_file['f_lower'][:]
+tid = numpy.arange(len(f_lower))
+mass1, mass2, spin1z, spin2z = triggers.get_mass_spin(temp_file, tid)
+chi_eff = convert.chi_eff(mass1, mass2, spin1z, spin2z)
```

Comment:
I don't know why you are still calculating these values here (template chi_eff, mchirp, eta).  If they are needed to make the kde then they should be calculated as a parameter in the loop over args.fit_param later. If they are not used in making the kde they do not have to be calculated.